host/distro/devtree.go

```diff
@@ -35,11 +41,33 @@ func DTCompatible() []string {
 	return dtCompatible
 }
 
+// DTRevision returns the device revision (e.g. a02082 for the Raspberry Pi 3)
+// from the Linux device tree, or the empty string if the file is missing or
+// malformed.
+func DTRevision() string {
+	mu.Lock()
+	defer mu.Unlock()
+
+	if dtRevision != "" {
+		return dtRevision
+	}
+	b, err := ioutil.ReadFile("/proc/device-tree/system/linux,revision")
+	if err != nil {
+		return ""
+	}
+	if len(b) < 4 {
+		return ""
+	}
+	dtRevision = fmt.Sprintf("%04x", binary.BigEndian.Uint32(b[:4]))
```

Comment:
Do you think it's always going to be a uint32? If so maybe we could keep it as a uint32.

One unstated part of the API here is that the format will always be prefixed with at least 4 zeros, and it's now implicitly baked into the expectations, e.g.

```
switch distro.DTRevision() {
  case "0x0001":
  cast "0x123456":
}
```

so this will be a breaking change to change the format string later. That's not an issue if you make it a uint32 or uint64.

Reply:
Done.